service/core/kafka/server/src/main/scala/com/lightbend/lagom/internal/broker/kafka/Producer.scala

```diff
@@ -121,12 +128,12 @@ private[lagom] object Producer {
         context.stop(self)
     }
 
-    private def run(tag: String, serviceLocatorUri: Option[URI], dao: OffsetDao) = {
+    private def run(tag: String, endpoints: String, dao: OffsetDao) = {
```

Comment:
The usage of `Option` could be avoid much higher on the stack. At the place we detect that no service lookup was done and that we must use `kafkaConfig.brokers`. 

The previous code was using `Either` and `Option` to make the decision on when of use the values from the service locator or from `kafkaConfig.brokers`.